prov/tcp/src/xnet_cq.c

```diff
@@ -148,9 +148,17 @@ void xnet_report_success(struct xnet_xfer_entry *xfer_entry)
 		return;
 	}
 
+
 	flags = xfer_entry->cq_flags & ~FI_COMPLETION;
 	if (flags & FI_RECV) {
 		len = xnet_msg_len(&xfer_entry->hdr);
+		if (xfer_entry->mrecv) {
```

Comment:
Wait -- this check doesn't work since moving mrecv into the union.  The checks like this may need to be updated to something like : xfer_entry->ctrl_flags & FI_MULTI_RECV

Reply:
meant XNET_MULTI_RECV for the ctrl_flag... it's the same value as FI_MULTI_RECV, but I'd use the internal definition instead

Reply:
yeah, it was in the back of my mind.  v1 guarantees mrecv but not v2.  There are 2 possible cases where mrecv is null, one, obviously is the failed allocation (unlikely) and the second is for the first receive to consume the RX buffer beyond the minimum.  I'm guessing it is the second case that's causing the failure.  I will make a change to accommodate these.